package proto_ff;

// calm 自定义的fieldoption都在该文件中
import "yd_fieldoptions.proto";
import "common_logic.proto";

//下线标记
enum LOGOUT_FLAG
{
  LOGOUT_FLAG_NULL           	= 0;					// NULL
  LOGOUT_FLAG_NORMAL         	= 1;					// 正常下线
  LOGOUT_FLAG_REPLACE        	= 2;					// 别处登陆，被踢下线
  LOGOUT_FLAG_SWITCH_SERVER  	= 3;					// 切换服务器下线
  LOGOUT_FLAG_BREAK_TIMEOUT  	= 4;					// 断线超时
  LOGOUT_FLAG_OPERATE_KICK	= 5;					// 管理后台T人
  LOGOUT_FLAG_CHEAR_KICK		= 6;					// 作弊被T
  LOGOUT_FLAG_KICK			= 7;					// 其它原因被T
  LOGOUT_FLAG_CRASH			= 8;					// 逻辑服挂了
  LOGOUT_FLAG_RECONNECT		= 9;					// 重连,关闭旧的连接
}

//返回码（公用的返回码放到前面，模块相关的返回码放到后面）
enum ERetCode
{
  option allow_alias = true;
  //公用返回码 0 - 99
  RET_SUCCESS = 0;  //成功
  RET_FAIL = 1;  //失败
}





